pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator_test.go

```diff
@@ -135,3 +135,117 @@ func TestFindAndAddActivePods_FindAndRemoveDeletedPods(t *testing.T) {
 	}
 
 }
+
+func TestFindAndRemoveDeletedPodsInFailedNodes(t *testing.T) {
```

Comment:
We should try and get some e2e tests for this. IIRC - this code path broke before too..(but for different reason).